src/net/dmytro/javapro/homework5/task2/obstacle/Obstacle.java

```diff
@@ -0,0 +1,10 @@
+package net.dmytro.javapro.homework5.task2.obstacle;
+
+import net.dmytro.javapro.homework5.task2.participant.Participant;
+
+public abstract class Obstacle {
+
+    public abstract void doIt(Participant liveParticipant);
```

Comment:
confusing parameter name,  if I read it I think that only live participants can doIt.